### ISSUE TITLE
Semicolon rule doesn't check interfaces

### ISSUE THREAD

**Issue by gscshoyru** [palantir/tslint#254]
```typescript
interface ITest {
    foo: string
}
```

should trigger missing semi-colon, but doesn't.

**Comment by awerlang** [comment]
The main reason against ASI is avoiding gotchas with expressions and statements. It is not a concern in class and interfaces declarations.

And according to http://www.typescriptlang.org/Content/TypeScript%20Language%20Specification.pdf, semicolons are separators and not terminators. But I think even tsc doesn't handle that as per spec.

```
TypeMemberList:
    TypeMember
    TypeMemberList ; TypeMember
```

**Comment by goldcaddy77** [comment]
Note: I found a similar (but different) semicolon linting issue #365
